index/index.go

```diff
@@ -107,11 +114,26 @@ func (index *Index) Get(key string) (*io.SectionReader, error) {
 	}
 
 	fileIndex, offset := deserializeIndexEntry(bytes)
-	index.readLocks[fileIndex].Lock()
-	defer index.readLocks[fileIndex].Unlock()
+	f := index.files[fileIndex]
+
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	_, err = f.file.Seek(offset, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	ok := f.reader.Scan()
+	if !ok {
+		if f.reader.Err() != nil {
+			return nil, f.reader.Err()
+		} else {
+			return nil, io.ErrUnexpectedEOF
```

Comment:
It feels a little weird that you're synthesizing an error here. Why doesn't the error get returned from `Scan()`?